test/src/unit-capi-vfs.cc

```diff
@@ -970,3 +979,89 @@ TEST_CASE_METHOD(VFSFx, "C API: Test VFS parallel I/O", "[capi][vfs]") {
     check_vfs(MEMFS_TEMP_DIR);
   }
 }
+
+TEST_CASE("C API: VFS recursive ls", "[capi][vfs][ls-recursive]") {
+  using namespace tiledb;
+  tiledb_ctx_t* ctx;
+  tiledb_vfs_t* vfs;
+  auto fs_vec = vfs_test_get_fs_vec();
+  REQUIRE(vfs_test_init(fs_vec, &ctx, &vfs).ok());
+  int64_t max_paths = GENERATE(10, 50, -1);
```

Comment:
This `GENERATE` declaration needs documentation, at minimum.

If `max_paths` argument is < 0, it's invalid. That test should be separated out in argument validation.

Reply:
I had used -1 here to request all results, similar to `VFS::ls_with_sizes`. I missed writing some docstrings for this parameter though so added them in today. I could use 0 instead with uin64_t, if that's preferred.

I got to most of the feedback here today, aside from the unresolved comments related to testing.